engine/flink/process/src/main/scala/pl/touk/nussknacker/engine/process/FlinkStreamingProcessRegistrar.scala

```diff
@@ -55,8 +55,10 @@ import scala.util.{Failure, Success}
 
 class FlinkStreamingProcessRegistrar(compileProcess: (EspProcess, ProcessVersion) => (ClassLoader) => CompiledProcessWithDeps,
                                      eventTimeMetricDuration: FiniteDuration,
-                                     checkpointInterval: FiniteDuration,
-                                     enableObjectReuse: Boolean, diskStateBackend: Option[StateBackend])
+                                     @Deprecated checkpointInterval: Option[FiniteDuration],
+                                     enableObjectReuse: Boolean,
+                                     diskStateBackend: Option[StateBackend],
+                                     checkpointConfig: Option[CheckpointConfig] = None)
```

Comment:
let's use only CheckpointConfig here, and handle legacy method only around line 174 (parsing config)